File: problemas/iniciante/2168/2168.py
mapa = []


def segura(x, y):
    return mapa[x][y] + mapa[x + 1][y] + mapa[x][y + 1] + mapa[x + 1][y + 1] >= 2


N = int(input())
for _ in range(N + 1):
    mapa.append([int(x) for x in input().strip().split(' ')])

for i in range(N):
    for j in range(N):
        print('S' if segura(i, j) else 'U', end='')
    print('')

File: problemas/iniciante/2168/2168.ts
process.stdin.resume();
process.stdin.setEncoding('utf8');

let input = '';
let lineCount = 0;

process.stdin.on('data', (chunk) => {
  input += chunk;
});

process.stdin.on('end', () => {
  var lines = input.split("\n");

  let mapa: number[][] = [];
  const segura = (x: number, y: number) => mapa[x][y] + mapa[x + 1][y] + mapa[x][y + 1] + mapa[x + 1][y + 1] >= 2;

  let N = parseInt(lines.shift()!);
  for (let i = 0; i < N + 1; ++i) {
    mapa.push(lines.shift()!.trim().split(' ').map(x => parseInt(x)));
  }

  let linha = '';
  for (let i = 0; i < N; ++i) {
    linha = '';
    for (let j = 0; j < N; ++j) {
      linha += (segura(i, j) ? 'S' : 'U');
    }
    console.log(linha);
  }
});